src/NSubstitute/Core/Arguments/ArgumentFormatter.cs

```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Linq;
 using System.Reflection;
+using NSubstitute.Extensions;
```

Comment:
I'm a bit worried about how easy this will be to get wrong (inadvertently breaking the NET40 build).

My IDE reports this as unused, it would be very easy to remove it and break the build. I thought about
surrounding these using NSubstitute.Extensions; lines with #if NET40 / #endif to make it more obvious, but this is going to cause a problem for new files (need to remember to include it/remove it whenever a call to is GetTypeInfo() added/removed).

Then I thought about creating a GetTypeInfoCompat() extension that will work for NET40 and others (using conditional compilation), but then we have the problem that contributors need to know to call that instead of the standard GetTypeInfo(). (Don't know if this is something we can build a Roslyn inspection-ey thing for? Would be nice if someone tried to compile and got a "Use GetTypeInfoCompat() instead " error.)

Any thoughts on this?

Reply:
Personally - I think I'm used to seeing this in multi-targeting projects. In my own project - I'd prefer the 'false alarm unused' over unnecessary pre-compilations, on the basis that the build will fail if anything is inadvertantly removed - but happy to update them as you'd prefer. 😄